OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,314 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds a particle-based overlay.")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("Factor for particle density. A smaller value increases the number of particles spawned.")]
+		public readonly int ParticleDensityFactor = 1311;
+
+		[Desc("The wind will change over time by the values of WindStrength, or just stick to the first value.")]
+		public readonly bool ChangingWind = true;
+
+		[Desc("Works only if ChangingWind is enabled. The levels of wind intensity desc. the x-movement value of the particles in px/tick.")]
+		public readonly int[] WindStrength = { -5, -3, -2, 0, 2, 3, 5 };
+
+		[Desc("Works only if ChangingWind is enabled. Min. and max. ticks needed to change the wind. ")]
+		public readonly int[] WindTick = { 150, 750 };
+
+		[Desc("The wind is fading softly from zero or instant.")]
+		public readonly bool InstantWindChanges = false;
+
+		[Desc("Particles are drawn with rectangles when enabled, else with lines.")]
+		public readonly bool UseRectangles = true;
+
+		[Desc("Works only with rectangles enabled. Size min. and max. value in pixels.")]
+		public readonly int[] PaticleSize = { 1, 3 };
+
+		[Desc("Scatters falling direction on the x-axis. Scatter min. and max. value in px/tick.")]
+		public readonly int[] ScatterDirection = { -1, 1 };
+
+		[Desc("The value that particles fall down. Gravity min. and max. value in px/tick.")]
+		public readonly float[] Gravity = { 1.00f, 2.00f };
+
+		[Desc("The current offset value for the swing movement. SwingOffset min. and max. value in px/tick.")]
+		public readonly float[] SwingOffset = { 1.0f, 1.5f };
+
+		[Desc("The value that particles swing to the side each update. SwingSpeed min. and max. value in px/tick.")]
+		public readonly float[] SwingSpeed = { 0.001f, 0.025f };
+
+		[Desc("The value range that can be swung to the left or right. SwingAmplitude min. and max. value in px/tick.")]
+		public readonly float[] SwingAmplitude = { 1.0f, 1.5f };
+
+		[Desc("The ArgbColors for the particles. Think about using the colors of the temperate.pal for example.")]
+		public readonly Color[] ParticleColors = {
+			Color.FromArgb(255, 236, 236, 236),
+			Color.FromArgb(255, 228, 228, 228),
+			Color.FromArgb(255, 208, 208, 208),
+			Color.FromArgb(255, 188, 188, 188)
+		};
+
+		[Desc("Works only with line enabled and will fade out the tail of the line.")]
```

Comment:
Contrails can have curves. Weather particles have not. both are lines, true. But besides that its like mentioning the fog of war because it's also transparent. So i guess the word "transparent" is missing in the Desc.